desktop-widgets/templateedit.cpp

```diff
@@ -43,7 +50,20 @@ TemplateEdit::TemplateEdit(QWidget *parent, const print_options &printOptions, t
 
 	ui->plainTextEdit->setPlainText(grantlee_template);
 	editingCustomColors = false;
+#ifdef USE_WEBENGINE
+	QWebEngineView *view;
+	preview = new Printer(nullptr, printOptions, newTemplateOptions, Printer::PREVIEW, false, this);
+	view = preview->webView;
+	int w = ui->label->width();
+	int h = ui->label->height();
+	view->resize(w, h);
+	preview->print();
+	view->setZoomFactor(0.25);
+	view->show();
+	ui->verticalLayout_3->replaceWidget(ui->label, view);
+#else
```

Comment:
why not have this in updatePreview?
it seems that webengine requires "preview" to persist as a pointer for some reason, while for the webkit case we can print the result to a pixmap for a Printer object on the stack. [3]

Reply:
The whole pain in this was coming form the fact that a QWebEngineView uses GL and thus cannot render to a QPixmap. The preview widget is now a full blown web browser that in Printer::print is told to render a local file (and upon Apply to render a new file).

Reply:
i think offscreen rendering should be still possible. according to this post it is a matter of syncing the loaded page (onViewLoaded):
https://stackoverflow.com/questions/53524652/is-it-possible-to-render-qwebenginepage-qwebengineview-offscreen

the QPainter target in that case is QImage but can also be a QPixmap.

Reply:
Thinking about this a bit more, I am now convinced that the whole QPixmap/QImage whatever thing is an unnecessary indirection and not having it but having the actual rendering thing is an advantage.

Reply:
in the original, i think the idea is to render to a bitmap offscreen and to not show the QWebView widget in the dialog, because it would make the preview interactive in a weird way.
i think we should avoid that in the new version as well either use a bitmap again or lock the widget for clicks and key presses entirely.

performance wise, showing the full engine widget on screen might be slow, but rendering to bitmap offscreen is also slow, so not clear what is the winner in that regard.

my vote goes for using a bitmap, as that is the method i've seen elsewhere, but leaving the decision to your best judgement.